.github/workflows/unit-test.yml

```diff
@@ -180,6 +185,8 @@ jobs:
         run: |
           if [[ "$PIPELINE_MODE" =~ .*"benchmarks".* ]]; then
             echo "args=$PYTEST_BENCHMARKS_ARGS $PYTEST_ADDITIONAL_ARGS" >> $GITHUB_OUTPUT
+          elif [[ "$PIPELINE_MODE" =~ .*"warnings".* ]]; then
+            echo "args=$PYTEST_ADDITIONAL_ARGS --continue-on-collection-errors" >> $GITHUB_OUTPUT
```

Comment:
Why not just add the inline flag to the `pytest_additional_args` input variable based on the warning level? As far as I can tell, this is the only step that uses the new pipeline mode in a meaningful way. If we get rid of it, we can also get rid of the new pipeline mode entirely.

We could even include the `-W` flag in the additional args, which would allow us to get rid of both the new pipeline mode, and no longer require the warning level as input to the `unit-test` workflow.

Reply:
Thanks for the suggestion. I'll take this as something to improve upon later